llama-index-integrations/indices/llama-index-indices-managed-llama-cloud/llama_index/indices/managed/llama_cloud/base.py

```diff
@@ -544,20 +922,47 @@ def upload_file(
             )
             if verbose:
                 print(f"Uploaded file {file.id} with name {file.name}")
-        if resource_info:
-            self._client.files.update(file_id=file.id, request=resource_info)
+
         # Add file to pipeline
         pipeline_file_create = PipelineFileCreate(file_id=file.id)
-        self._client.pipelines.add_files_to_pipeline(
+        self._client.pipelines.add_files_to_pipeline_api(
             pipeline_id=self.pipeline.id, request=[pipeline_file_create]
         )
 
         if wait_for_ingestion:
-            self._wait_for_file_ingestion(
-                file.id, verbose=verbose, raise_on_error=raise_on_error
+            self.wait_for_completion(
+                file_ids=[file.id], verbose=verbose, raise_on_error=raise_on_error
             )
         return file.id
 
+    async def aupload_file(
+        self,
+        file_path: str,
+        verbose: bool = False,
+        wait_for_ingestion: bool = True,
```

Comment:
should we rename to `wait_for_completion` just for consistency with the new method name?